ddl/table.go

```diff
@@ -380,6 +396,26 @@ func checkTableNotExists(t *meta.Meta, job *model.Job, schemaID int64, tableName
 	return nil
 }
 
+// checkTableNameChange checks whether the table name has been changed, in cases where both sql1 and sql2 execute in parallel,
+// if sql1 is a renameTable it may execute successfully first, and sql2 may get the old table name, then this is not the desired result,
+// executing the renameTable DDL job does not change the tableID ,sql1 and sql2 were executed successfully,the result was not expected,
+// for example: sql1: "alter table t rename to t_add;"  sql2: "alter table t add a1 int;".
+func checkTableNameChange(t *meta.Meta, job *model.Job, tableName string, originalTableName string) error {
+	dbInfo, err := t.GetDatabase(job.SchemaID)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// the old version of tidb will not check this error and will execute successfully,
+	// the new version of tidb checks for errors and prompts for errors and cancels job.
```

Comment:
"The originalTableName will be nil when the job is enqueued by old version TiDB"